### DISCUSSION TITLE
device constantly restarting. (Guru Meditation Error: Core  0 panic'ed )

### DISCUSSION THREAD

**Discussion by maiaz** [jomjol/AI-on-the-edge-device#1354] [Q&A] [answered]
i got bunch of esp32 cam to read my analog meters, flashed, written SD card and installed in place. They was working OK before configuration was done. after i completed configuration, both devices start constantly rebooting after this lines in console log:

```
I (11339) sntp: The current date/time in Berlin is: Tue Nov 15 15:47:02 2022
I (11339) sntp: The current date/time in Berlin is: 2022-11-15_15:47
                                                                        nternal Flash-LED turn off PWM
Image loaded from memory: 640, 480, 3
save to file: /sdcard/log/source/20221115/15/raw_20221115-154845.jpg
I (26119) flow_make_image: remove old log images
I (26129) flow_make_image: 0 older log files deleted. 2 current log files not deleted.
        FlowControll.doFlow - ClassFlowAlignment
Vor ft->FindTemplate(_temp1);  /sdcard/config/ref0.jpg
Vor ft->FindTemplate(_temp2);  /sdcard/config/ref1.jpg
Alignment: dx -1073583021 - dy -1073490981 - rot 114.167328
2022-11-15T15:49:00
        FlowControll.doFlow - ClassFlowCNNGeneral
General 0 - Align&Cut
Guru Meditation Error: Core  0 panic'ed (LoadProhibited). Exception was unhandled.

Core  0 register dump:
PC      : 0x40196f5b  PS      : 0x00060030  A0      : 0x801955c4  A1      : 0x3fffb140
A2      : 0x3ffc3888  A3      : 0x00000057  A4      : 0x00000133  A5      : 0x00000015
A6      : 0x00000155  A7      : 0x00000000  A8      : 0x00000022  A9      : 0x3fffae10
A10     : 0x00000016  A11     : 0xffffffff  A12     : 0x00000001  A13     : 0x3fffb020
A14     : 0x00000016  A15     : 0x3fffb020  SAR     : 0x0000000b  EXCCAUSE: 0x0000001c
EXCVADDR: 0x0000002c  LBEG    : 0x4008c729  LEND    : 0x4008c739  LCOUNT  : 0xffffffff

Backtrace:0x40196f58:0x3fffb140 0x401955c1:0x3fffb170 0x401957fe:0x3fffb220 0x4019581e:0x3fffb240 0x4018c5bf:0x3fffb280 0x400ff7b9:0x3fffb340 0x400ffc15:0x3fffb390


ELF file SHA256: 28356ed42df4c694

Rebooting...
```

my config.ini below (not sure about it as it was wining about some parameters incorrect, but it was unable to set anything there, disabled input).

```
[MakeImage]
LogImageLocation = /log/source
WaitBeforeTakingPicture = 5
LogfileRetentionInDays = 15
Brightness = 0
Contrast = 0
Saturation = 0
LEDIntensity = 30
ImageQuality = 12
ImageSize = VGA
FixedExposure = false

[Alignment]
InitialRotate = 181
InitialMirror = false
SearchFieldX = 20
SearchFieldY = 20
AlignmentAlgo = default
FlipImageSize = false
/config/ref0.jpg 103 271
/config/ref1.jpg 442 142

[Digits]
Model = 
CNNGoodThreshold = 0.5
;LogImageLocation = /log/digit
;LogfileRetentionInDays = 3
main.d1 87 307 21 34 0
main.d2 123 307 21 34 0
main.d3 163 307 21 34 0
main.d4 201 307 21 34 0
main.d5 241 307 21 34 0
decimals.dd1 283 305 25 43 0
decimals.dd2 315 305 25 43 0
decimals.dd3 362 305 25 43 0

;[Analog]
Model = 
;LogImageLocation = /log/analog
;LogfileRetentionInDays = 3
main.ana1 432 230 92 92 false
main.ana2 379 332 92 92 false
main.ana3 283 374 92 92 false
main.ana4 155 328 92 92 false

[PostProcessing]
main.DecimalShift = 0
;decimals.DecimalShift = 
main.AnalogDigitalTransitionStart = 9.2
;decimals.AnalogDigitalTransitionStart = 
PreValueUse = true
PreValueAgeStartup = 720
AllowNegativeRates = false
main.MaxRateValue = 0.05
ErrorMessage = true
CheckDigitIncreaseConsistency = false

LEDType = WS2812
LEDNumbers = 2
LEDColor = 150 150 150

[AutoTimer]
AutoStart = true
Intervall = 4.85

[Debug]
Logfile = 3
LogfileRetentionInDays = 3

[System]
TimeZone = CET-1CEST,M3.5.0,M10.5.0/3
SetupMode = false
```

any idea what could be wrong for me?

**Comment by jomjol** [comment]
There is a problem with the alignment:
```Alignment: dx -1073583021 - dy -1073490981 - rot 114.167328```
This large negative numbers cannot be. Is it possible, that your mark is at the outer border of the image? Or you reference pictures (ref0, ref1) has a size of 1x1 or similar?

**Reply by maiaz** [reply]
initial problem was that Model was missing in config, during configuration it leaves selection box empyt (js error?) and then restarts. after i put it into config.ini restarts stopped, but still not possible to operate. first of all reference pictures do not exists at SD card. second - when it works it constantly trigger watchdog for http so it is not possible to do anything with web gui..  i believe that somehow related to configuration done wrong.  also some files is missing on flash (cpu_temperature, etc) so.. trying to figure out what went wrong. still not working for me.

**Comment by maiaz** [comment]
trying to follow comfiguration process... seeing a lot of weird errors.. are that expected?
```
readconfigcommon.js:198          POST http://192.168.12.252/delete/img_tmp/ref0.jpg 400 (Bad Request)
readconfigcommon.js:171          GET http://192.168.12.252/editflow?task=copy&in=/config/ref0.jpg&out=/img_tmp/ref0.jpg 500 (Internal Server Error)
readconfigcommon.js:198          POST http://192.168.12.252/delete/img_tmp/ref0_org.jpg 400 (Bad Request)
readconfigcommon.js:171          GET http://192.168.12.252/editflow?task=copy&in=/config/ref0_org.jpg&out=/img_tmp/ref0_org.jpg 500 (Internal Server Error)
readconfigcommon.js:198          POST http://192.168.12.252/delete/img_tmp/ref1.jpg 400 (Bad Request)
readconfigcommon.js:171          GET http://192.168.12.252/editflow?task=copy&in=/config/ref1.jpg&out=/img_tmp/ref1.jpg 500 (Internal Server Error)
readconfigcommon.js:198          POST http://192.168.12.252/delete/img_tmp/ref1_org.jpg 400 (Bad Request)
readconfigcommon.js:171          GET http://192.168.12.252/editflow?task=copy&in=/config/ref1_org.jpg&out=/img_tmp/ref1_org.jpg 500 (Internal Server Error)
/fileserver/img_tmp/ref0.jpg?1668633654216:1          GET http://192.168.12.252/fileserver/img_tmp/ref0.jpg?1668633654216 404 (Not Found)
ref0_org.jpg:1          GET http://192.168.12.252/fileserver/img_tmp/ref0_org.jpg?1668633654216 404 (Not Found)
```

and same time in console:
```
uri: /editflow?task=copy&in=/config/ref0_org.jpg&out=/img_tmp/ref0_org.jpg
1 uri: /editflow?task=copy&in=/config/ref0_org.jpg&out=/img_tmp/ref0_org.jpg, filename: , filepath: /sdcard
Filename:
File requested: /sdcard/html/editflow
open file /sdcard/html/editflow in mode r
Error: file /sdcard/html/editflow does not exist!
E (107949) serverhelp: Failed to read existing file : /sdcard/html/editflow
W (107949) httpd_txrx: httpd_resp_send_err: 500 Internal Server Error - Failed to read existing file
W (107969) httpd_uri: httpd_uri: uri handler execution failed
delete_post_handler
E (107999) file_server: File does not exist : /img_tmp/ref1.jpg
W (107999) httpd_txrx: httpd_resp_send_err: 400 Bad Request - File does not exist
W (107999) httpd_uri: httpd_uri: uri handler execution failed
uri: /editflow?task=copy&in=/config/ref1.jpg&out=/img_tmp/ref1.jpg
1 uri: /editflow?task=copy&in=/config/ref1.jpg&out=/img_tmp/ref1.jpg, filename: , filepath: /sdcard
Filename:
File requested: /sdcard/html/editflow
open file /sdcard/html/editflow in mode r
Error: file /sdcard/html/editflow does not exist!
E (108059) serverhelp: Failed to read existing file : /sdcard/html/editflow
W (108059) httpd_txrx: httpd_resp_send_err: 500 Internal Server Error - Failed to read existing file
W (108069) httpd_uri: httpd_uri: uri handler execution failed
delete_post_handler
E (108109) file_server: File does not exist : /img_tmp/ref1_org.jpg
W (108109) httpd_txrx: httpd_resp_send_err: 400 Bad Request - File does not exist
W (108119) httpd_uri: httpd_uri: uri handler execution failed
uri: /editflow?task=copy&in=/config/ref1_org.jpg&out=/img_tmp/ref1_org.jpg
1 uri: /editflow?task=copy&in=/config/ref1_org.jpg&out=/img_tmp/ref1_org.jpg, filename: , filepath: /sdcard
Filename:
File requested: /sdcard/html/editflow
open file /sdcard/html/editflow in mode r
Error: file /sdcard/html/editflow does not exist!
E (108179) serverhelp: Failed to read existing file : /sdcard/html/editflow
W (108179) httpd_txrx: httpd_resp_send_err: 500 Internal Server Error - Failed to read existing file
W (108199) httpd_uri: httpd_uri: uri handler execution failed
```


is that expected?

**Reply by caco3** [reply]
`Error: file /sdcard/html/editflow does not exist!`

This error also got reported by other users, see https://github.com/jomjol/AI-on-the-edge-device/discussions/1346

Can you please let us know which version you are using?

**Reply by maiaz** [reply]
Branch: 'HEAD', Tag: 'v12.0.1', Revision: dfe2c46, Date/Time: 2022-09-29 17:13

flashed here with web-flasher.

**Comment by maiaz** [comment, marked as answer]
so, i compiled everything from scratch, and now on rolling / https://github.com/jomjol/AI-on-the-edge-device/commit/e15ea561bceeaa80d9b5d3688bbc83c59a1def7f+ / 2022-11-17 15:35
no editflow errors, much less errors in browser console, and finally it is possible to use web GUI now.. other hassle with image alignement, but that's another story. looks like something is wrong with version which is fed via web installer.

**Reply by maiaz** [reply]
Git revision: e15ea56+

**Reply by caco3** [reply]
Are you aware you could have used the auto build rolling?
See https://github.com/jomjol/AI-on-the-edge-device/actions?query=branch%3Arolling

The firmware.bin used for the Web installer is identical to the firmware.bin in the release https://github.com/jomjol/AI-on-the-edge-device/releases/tag/v12.0.1 (same md5sum).

So if it did not work with the Web Installer, it either is a coincidence or the Web Installer does something wrong. The later is very unlikely:
 - The Web installer is not written by us and used on many other projects
 - If it would corrupt the firmware during flashing, it is very likely that the firmware would not work at all

So, could you, if possible test once more with a manual flashing using the files in the release https://github.com/jomjol/AI-on-the-edge-device/releases/tag/v12.0.1 ?

**Reply by maiaz** [reply]
i still have a bunch of non critical issues with rolling release - it wining about gui version mismatch with FW, also common.js looks like checking version incorrectly (or get it in format which is not expected). i still have 3rd esp32-cam so i will try that option you asked.

**Reply by jomjol** [reply]
please check the newest rolling - there should be some bugs removed (e.g. version controll)

**Reply by Jake-a-Lake** [reply]
I am having this issue as well and after a successful build of the newest rolling and upload. It appears the device is constantly rebooting.  
[jomol_rolling_log.txt](https://github.com/jomjol/AI-on-the-edge-device/files/10047241/jomol_rolling_log.txt)

**Reply by maiaz** [reply]
@Jake-a-Lake in your case you need external power supply, it goes to reboot due to insufficient power when it try to kick in WiFi. 

`Brownout detector was triggered

ets Jun  8 2016 00:22:57
`
I fixed that by connecting 5V to 5V and GND pin from external PS, and removed 3V3 (or 5V) pin from FTDI (kept only rx/tx/GND).

**Reply by maiaz** [reply]
> So, could you, if possible test once more with a manual flashing using the files in the release https://github.com/jomjol/AI-on-the-edge-device/releases/tag/v12.0.1 ?

i just took my 3rd device, flashed with web installed again, copied sd-card contents, and get into same error - after new reference image created and saved, after it is impossible to set any alignement references, same error about editflow:

**browser inspector:**

https://i.imgur.com/YpcsdfP.png



**console:**

uri: /editflow?task=copy&in=/config/ref0.jpg&out=/img_tmp/ref0.jpg
1 uri: /editflow?task=copy&in=/config/ref0.jpg&out=/img_tmp/ref0.jpg, filename: , filepath: /sdcard
Filename:
File requested: /sdcard/html/editflow
open file /sdcard/html/editflow in mode r
Error: file /sdcard/html/editflow does not exist!
E (93919) serverhelp: Failed to read existing file : /sdcard/html/editflow
W (93919) httpd_txrx: httpd_resp_send_err: 500 Internal Server Error - Failed to read existing file
W (93939) httpd_uri: httpd_uri: uri handler execution failed
delete_post_handler
E (93979) file_server: File does not exist : /img_tmp/ref0_org.jpg
W (93979) httpd_txrx: httpd_resp_send_err: 400 Bad Request - File does not exist
W (93989) httpd_uri: httpd_uri: uri handler execution failed
uri: /editflow?task=copy&in=/config/ref0_org.jpg&out=/img_tmp/ref0_org.jpg
1 uri: /editflow?task=copy&in=/config/ref0_org.jpg&out=/img_tmp/ref0_org.jpg, filename: , filepath: /sdcard
Filename:
File requested: /sdcard/html/editflow
open file /sdcard/html/editflow in mode r
Error: file /sdcard/html/editflow does not exist!
E (94089) serverhelp: Failed to read existing file : /sdcard/html/editflow
W (94089) httpd_txrx: httpd_resp_send_err: 500 Internal Server Error - Failed to read existing file
W (94109) httpd_uri: httpd_uri: uri handler execution failed
delete_post_handler
E (94179) file_server: File does not exist : /img_tmp/ref1.jpg
W (94179) httpd_txrx: httpd_resp_send_err: 400 Bad Request - File does not exist
W (94189) httpd_uri: httpd_uri: uri handler execution failed
uri: /editflow?task=copy&in=/config/ref1.jpg&out=/img_tmp/ref1.jpg
1 uri: /editflow?task=copy&in=/config/ref1.jpg&out=/img_tmp/ref1.jpg, filename: , filepath: /sdcard
Filename:
File requested: /sdcard/html/editflow
open file /sdcard/html/editflow in mode r
Error: file /sdcard/html/editflow does not exist!
E (94299) serverhelp: Failed to read existing file : /sdcard/html/editflow
W (94299) httpd_txrx: httpd_resp_send_err: 500 Internal Server Error - Failed to read existing file
W (94319) httpd_uri: httpd_uri: uri handler execution failed
delete_post_handler

or idea was to take firmware.bin and u pload it manually w/o web flasher?

**Reply by caco3** [reply]
In https://github.com/jomjol/AI-on-the-edge-device/discussions/1336#discussioncomment-4224496 someone reported a similar issue ("Failed to read existing file"). The issue seemed to be a SD-Card which seems not to work well with the ESP32-CAM.

**Reply by maiaz** [reply]
> In [#1336 (comment)](https://github.com/jomjol/AI-on-the-edge-device/discussions/1336#discussioncomment-4224496) someone reported a similar issue ("Failed to read existing file"). The issue seemed to be a SD-Card which seems not to work well with the ESP32-CAM.

could be, but its weird that update firmware solved for me with a same SD card.
